packages/@aws-cdk/aws-globalaccelerator/lib/accelerator.ts

```diff
@@ -85,9 +86,16 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
   constructor(scope: Construct, id: string, props: AcceleratorProps = {}) {
     super(scope, id);
 
+    const maxResourceNameLength = 64;
+    const name = cdk.FeatureFlags.of(this).isEnabled(cxapi.GLOBAL_ACCELERATOR_SAFE_RESOURCE_NAME) ?
+      cdk.Names.uniqueResourceName(this, {
+        separator: '-',
+        maxLength: maxResourceNameLength,
+        allowedSpecialCharacters: '/_-',
+      }) : props.acceleratorName ?? cdk.Names.uniqueId(this);
```

Comment:
We'll still have a problem here if `acceleratorName` is longer than 64 characters. I'm not seeing anywhere that we validate the length of it. We should probably add that to this change.

Reply:
Should I trim the length automatically`(props.acceleratorName ?? cdk.Names.uniqueId(this)).slice(-64);` or check the value and throw an error in case it exceeds the character limit?